### ISSUE TITLE
using async function in a pre hook

### ISSUE THREAD

**Issue by kerolloz** [typegoose/typegoose#381]
```typescript
@pre<User>('save', async function () {
  this.password = await Password.hash(this.password);
})
export class User {
  @prop()
  name!: string;

  @prop()
  password!: string;
}
```
Gives the following error: 

`Promise returned in function argument where a void return was expected. `eslint([@typescript-eslint/no-misused-promises](https://github.com/typescript-eslint/typescript-eslint/blob/v4.2.0/packages/eslint-plugin/docs/rules/no-misused-promises.md))

Is there a way to fix this? or it needs to be changed in the TypeScript typings?
https://github.com/typegoose/typegoose/blob/26afc6933e834b42a6ac02920658a240c6eda18a/src/hooks.ts#L14
```diff
+  type PreFnWithDocumentType<T> = (this: DocumentType<T>, next: HookNextErrorFn) => void | Promise<void>; 
-  type PreFnWithDocumentType<T> = (this: DocumentType<T>, next: HookNextErrorFn) => void; 
```

## Versions

-  "@typegoose/typegoose": "^7.4.0"
-  "mongoose": "^5.10.6"
-  "typescript": "^4.0.3"

**Comment by hasezoey** [comment]
this definetly seems wrong, but i cant replicate it, could you provide an reproduction repo?

**Comment by kerolloz** [comment]
Yes, sure. 
https://github.com/kerolloz/typegoose-bug
Make sure ESLint is working in VSCode.  
![image](https://user-images.githubusercontent.com/36763164/93913571-92561400-fd05-11ea-9a36-c0787ca81620.png)

**Comment by hasezoey** [comment]
oh right forgot that, my project(s) dont use eslint (yet), so i couldnt reproduce it

will fix it

**Comment by kerolloz** [comment]
Thank you @hasezoey :heart_eyes:

**Comment by hasezoey** [comment]
will be included in 7.4.1

**Comment by kerolloz** [comment]
That's great! Thanks :sparkles:
